### PR TITLE
Order of courses

### PR DESCRIPTION
**Pull request by Christen97**
Changed the order of the courses on the dashboard, to make the newest courses at the top. 
Changed the order of the exercises on the dashboard, this way the dashboard shows the 5 newest exercises.

### CODE REVIEW

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -20,11 +20,11 @@ public function index() : View
     {
         $awaitingFeedback = auth()->user()->feedback()->with(['taskDelegation', 'project.task.course'])->unreviewed()->get()
             ->filter(fn(ProjectFeedback $feedback) => $feedback->taskDelegation->deadline_at->isFuture());
-        $courses = auth()->user()->courses()->get();
+        $courses = auth()->user()->courses()->orderBy('created_at', 'asc')->get();
```
Comment (jazerix) [inline app/Http/Controllers/DashboardController.php]: Skal det ikke være desc? Så det nyeste kursus altid står øverst

Comment (mcamp) [reply app/Http/Controllers/DashboardController.php]: remember to update the tests too

**Review by jazerix — APPROVED** [review]

**Comment by mcamp** [comment]
@Christen97 do we have tests?
If we do, just fix the conflict and formatting

**Comment by Christen97** [comment]
> @Christen97 do we have tests? If we do, just fix the conflict and formatting

I have made the test and it should be ready for merging now

File: tests/Feature/DashboardTest.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        'name'       => 'furthest assignment',
+        'ends_at'    => Carbon::now()->addMonth(),
+    ]);
+});
+
+it('tests courses are ordered by desc', function () {
+    $student = User::factory()->hasAttached([$this->oldCourse, $this->newCourse])->create();
+    actingAs($student);
+
+    $this->assertEquals('new course', $student->courses()->orderBy('created_at', 'desc')->first()->name);
```
Comment (jazerix) [inline tests/Feature/DashboardTest.php]: This is a unit test and not a feature test - you're not actually testing the dashboard

File: tests/Feature/DashboardTest.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+});
+
+it('tests exercises are ordered by desc', function () {
+    $student = User::factory()->hasAttached($this->oldCourse)->create();
+    actingAs($student);
+
+    $this->assertEquals('new exercise', Task::exercises()->whereIn('course_id', $this->oldCourse->pluck('id'))->orderBy('created_at', 'desc')->visible()->first()->name);
```
Comment (jazerix) [inline tests/Feature/DashboardTest.php]: Same here

File: tests/Feature/DashboardTest.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+});
+
+it('tests assignments are ordered by nearest deadline', function () {
+    $student = User::factory()->hasAttached($this->oldCourse)->create();
+    actingAs($student);
+
+    $this->assertEquals('nearest assignment', Task::assignments()->whereIn('course_id', $this->oldCourse->pluck('id'))->orderBy('ends_at', 'asc')->visible()->first()->name);
```
Comment (jazerix) [inline tests/Feature/DashboardTest.php]: same here

**Review by jazerix — CHANGES_REQUESTED** [review]

**Comment by jazerix** [comment]
You should probably look into something like: https://laravel.com/docs/9.x/http-tests#assert-see-in-order

**Review by jazerix — APPROVED** [review]

**Pull request merged**